### PR TITLE
drop support for lambda/<> and glue/<>

### PR DESCRIPTION
**Pull request by tnsardesai**
https://clever.atlassian.net/browse/INFRANG-4773

- improve logging for all 3 ways to publish catapult applications
- make publishing catapult applications strict. i.e launch.yml has to exist and $RUN_TYPE has to match the type of script that is called
- drop support for lambda/<region> runtype
- add support for glue runtype

Testing:
- successful glue test run https://app.circleci.com/pipelines/github/Clever/analytics-district-participation/567/workflows/e8af8d56-d09e-4741-8fc3-7f6aa04e71cf/jobs/2123 (failing due to a different bug, which I will fix in catapult in a bit)
- didn't do others as code change is straightforward

### CODE REVIEW

**Review by nbhatia823 — APPROVED** [review]

**Comment by tnsardesai** [comment]
@nbhatia823 I just learned that services can also have empty run type so I will be pushing another commit to this PR

**Review by nbhatia823 — APPROVED** [review]
Nice, this approach seems cleaner for getting run type. The last was pretty hard to follow, but it also was there before this PR. Did you change it just for readability or was there something also incorrect about the previous method?

**Comment by tnsardesai** [comment]
The previous approach relied on "grep type". So if an application had the word type more than once it breaks. Which is always the case for services. For example the output had multiple empty lines for catapult 
```
>grep type "launch/catapult.yml" | cut -d' ' -f4
docker



```
because grep type returns a lot of lines.
```
grep type "launch/catapult.yml"
  type: docker
    type: http
- type: InternalErrorAlarm
- type: InternalErrorAlarm
- type: InternalErrorAlarm
```
This new approach is cleaner and will also be correct all the time
